### PR TITLE
Improving code quality

### PR DESCRIPTION
**Pull request by alainlompo**

### CODE REVIEW

**Comment by thegedge** [comment]
Hey, @alainlompo. thanks for contributing. Sorry for the delay in reviewing, but I'll try to get to this asap (I'm quite busy lately!).

File: src/main/java/ca/gedge/radixtree/RadixTreeUtil.java [hunk 1]
```diff
@@ -21,10 +21,14 @@
...
 package ca.gedge.radixtree;
 
+import java.io.Serializable;
+
 /**
  * Radix tree utility functions.
  */
 public class RadixTreeUtil {
+	private RadixTreeUtil() {
+	}
```
Comment (thegedge) [inline src/main/java/ca/gedge/radixtree/RadixTreeUtil.java:31]: Empty line after this one

File: src/main/java/ca/gedge/radixtree/RadixTree.java [hunk 2]
```diff
@@ -480,7 +482,7 @@ private V remove(String key, RadixTreeNode<V> node) {
 			final int largestPrefix = RadixTreeUtil.largestPrefixLength(key, child.getPrefix());
 			if(largestPrefix == child.getPrefix().length() && largestPrefix == key.length()) {
 				// Found our match, remove the value from this node
-				if(child.getChildren().size() == 0) {
+				if(child.getChildren().isEmpty()) {
```
Comment (thegedge) [inline src/main/java/ca/gedge/radixtree/RadixTree.java:486]: 👌

File: src/main/java/ca/gedge/radixtree/RadixTree.java [hunk 3]
```diff
@@ -41,7 +41,9 @@
  * 
  * @param <V>  the type of values stored in the tree
  */
-public class RadixTree<V> implements Map<String, V>, Serializable {
+public class RadixTree<V extends Serializable> implements Map<String, V>, Serializable {
+	public static final String KEY_CANNOT_BE_NULL = "key cannot be null";
+	public static final String KEYS_MUST_BE_STRING_INSTANCES = "keys must be String instances";
```
Comment (thegedge) [inline src/main/java/ca/gedge/radixtree/RadixTree.java:46]: Empty line after this one

**Review by thegedge — COMMENTED** [review]
Sorry for the delayed review, @alainlompo, but I appreciate the PR. I have a few stylistic nitpicks, but otherwise we're good to merge :)

**Comment by alainlompo** [comment]
Hey @thegedge, thanks for reviewing. I did the changes :). More PRs soon

**Review by thegedge — APPROVED** [review]
Glad to hear you're interested in adding more PRs! I unfortunately don't do a lot of Java these days so I hardly find the time (or the motivation) to work on some of my old projects. Will be happy to review any PRs you send. Maybe it'll inspire me to push my own :)

**Pull request merged**